hipe4ml/tree_handler.py

```diff
@@ -136,22 +134,15 @@ def apply_preselections(self, preselections, inplace=True):
         inplace: bool
             If True, the preselected dataframe replaces the initial dataframe. Otherwise return a copy of the
             preselected df
+        **kwds: extra arguments are passed on to the pandas.DataFrame.query method:
+                https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html#pandas.DataFrame.query
 
         Returns
         ------------------------------------------------
         out: pandas.DataFrame or None
             If inplace == True return None is returned and the full DataFrame is replaced
-
         """
-        df_pres = None
-        if inplace:
-            self._preselections = preselections
-            self._full_data_frame = self._full_data_frame.query(
-                self._preselections)
-            return df_pres
-
-        df_pres = self._full_data_frame.query(preselections)
-        return df_pres
+        return self._full_data_frame.query(preselections, inplace=inplace, **kwds)
```

Comment:
Are you sure that this works also for ```inplace=True```? It looks to me that the pandas query function does not return a value when performing the operation in place https://github.com/pandas-dev/pandas/blob/v1.0.3/pandas/core/frame.py#L3083-L3243.

In addition, I think that storing the selection string in ```self._preselection``` when performing the operation in place could be useful.